web_src/js/features/repo-diff.js

```diff
@@ -103,4 +103,27 @@ export function initRepoDiffShowMore() {
       $('#diff-incomplete').replaceWith($(resp).find('#diff-file-boxes').children());
     });
   });
+  $(document).on('click', 'a.diff-show-more-button', (e) => {
+    e.preventDefault();
+    const $target = $(e.target);
+
+    if ($target.hasClass('disabled')) {
+      return;
+    }
+
+    $target.addClass('disabled');
```

Comment:
We should only disable it when loading successfully, if load failed, we should have the ability to try it again.

Reply:
Or should this be `loading`?

Reply:
I've added a fail handler to this and show more files button